### PR TITLE
Add check for owner references in backup sync, removing if missing

### LINKED ISSUE #7031: Backups not syncing to new cluster for migration if owner reference does not exist
**What steps did you take and what happened:**
I created a kubernetes cluster on Digital ocean to verify my migration and backup plans, and when attempting to validate my migration plan, the backups were recognized by velero and logged as imported, but do not show up in the cluster

**What did you expect to happen:**

velero backup describe <backup-name> to show the information for the backup that was logged as successfully synced to the cluster.

**The following information will help us better understand what's going on**:

_If you are using velero v1.7.0+:_  
Please use `velero debug  --backup <backupname> --restore <restorename>` to generate the support bundle, and attach to this issue, more options please refer to `velero debug --help` 

> An error occurred: backups.velero.io "<backupname>" not found

**Anything else you would like to add:**
The logs specifically show all of the pre-existing backups as successfully migrated ([this log line](https://github.com/vmware-tanzu/velero/blob/main/pkg/controller/backup_sync_controller.go#L184)), but none of the backups show up in any of the methods for checking for them. lens, kubectl, velero describe, etc.

**Environment:**

- Velero version: 1.12
- Velero features (use `velero client config get features`): NOT SET
- Kubernetes version (use `kubectl version`): 1.23 on old server, 1.28 on new
- Kubernetes installer & version: kubeadm install to digital ocean
- Cloud provider or hardware configuration: moving from local to digital ocean
- OS (e.g. from `/etc/os-release`): unknown

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

### PR DESCRIPTION
**Pull request by deefdragon**
# Please add a summary of your change
Loops through the list of Ownership References in a backup when synchronizing. For each ownership reference, it checks for the referenced schedule, and if it exists, keeps it, but if it does not exist, removes it, and logs as such.

This is only a draft as I am unsure if this is the best approach/inline with the desired usage of OwnershipReferences in velero. If a seperate approach is desired, I can update as requested.

# Does your change fix a particular issue?

Fixes #7031

# Please indicate you've done the following:

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [x] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

Will update docs as requested where relevant, but I don't think they are relevant anywhere in this particular case?

### CODE REVIEW

File: pkg/controller/backup_sync_controller.go [hunk 1]
```diff
@@ -171,6 +172,27 @@ func (b *backupSyncReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		}
 		backup.Labels[velerov1api.StorageLocationLabel] = label.GetValidName(backup.Spec.StorageLocation)
 
+		//check for the backup schedule. If it does not exist, remove it.
+		listedReferences := backup.ObjectMeta.OwnerReferences
+		foundReferences := make([]metav1.OwnerReference, 0)
```
Comment (ywk253100) [inline pkg/controller/backup_sync_controller.go:177]: How about only checking the schedule owner reference explicitly?
Although the schedule is the only owner of the backup currently, we may add more in the future.

Comment (deefdragon) [reply pkg/controller/backup_sync_controller.go:177]: do you mean just checking the owner reference type and only checking the schedules? if so I can add that yeah.

File: pkg/controller/backup_sync_controller.go [hunk 2]
```diff
@@ -171,6 +172,27 @@ func (b *backupSyncReconciler) Reconcile(ctx context.Context, req ctrl.Request)
...
+		for _, v := range listedReferences {
+			schedule := new(velerov1api.Schedule)
```
Comment (deefdragon) [inline pkg/controller/backup_sync_controller.go:179]: should I be checking That the uid of the fetched and stored schedules are the same? it appears the other issue specifically references uid as an issue.

**Review by deefdragon — COMMENTED** [review]
request for feedback here

File: pkg/controller/backup_sync_controller.go (hunk 1 above)
Comment (ywk253100) [reply pkg/controller/backup_sync_controller.go:177]: Yes

File: pkg/controller/backup_sync_controller.go (hunk 2 above)
Comment (ywk253100) [reply pkg/controller/backup_sync_controller.go:179]: Yes, check the UID either to make sure it's the same schedule

**Comment by deefdragon** [comment]
Updated the code to check a bit better, and account for ownership references.

I've tried writing some tests, but I'm having difficulty getting the tests to run, and I'm really not understanding how the existing testing works. I cant figure out how `Test Backup Sync Reconciler basic function` is validating that the end result is as expected.

**Comment by ywk253100** [comment]
> Updated the code to check a bit better, and account for ownership references.
> 
> I've tried writing some tests, but I'm having difficulty getting the tests to run, and I'm really not understanding how the existing testing works. I cant figure out how `Test Backup Sync Reconciler basic function` is validating that the end result is as expected.

How about splitting the newly added logic into a separate function so that you can add unit tests for the function easily?

**Comment by ywk253100** [comment]
@deefdragon Could you sign off the commit to pass the DCO checking?

**Comment by deefdragon** [comment]
sorry. work has kept me super busy the last week. I will get the tests in and signed off hopefully Saturday.

**Comment by ywk253100** [comment]
@deefdragon Do you have time to handle the PR now? As we are close to the FC date, if you don't have enough time, I'll take it over

**Comment by deefdragon** [comment]
I'll try to get to it tonight and clean it up/test it/sign it, but if I don't finish it tonight, feel free to take over. 

(I originally ran into this because I'm migrating my cluster, which is taking most of my attention. sorry for the delays.)

**Comment by deefdragon** [comment]
Should be signed, split out, and tested. I arguably went overboard on the tests, but I'd rather overdo it than miss something.

**Review by ywk253100 — APPROVED** [review]

**Review by blackpiglet — APPROVED** [review]

**Pull request merged**